runelite-client/src/main/java/net/runelite/client/plugins/emojis/EmojiPlugin.java

```diff
@@ -54,19 +77,52 @@
 @Slf4j
 public class EmojiPlugin extends Plugin
 {
-	private static final Pattern TAG_REGEXP = Pattern.compile("<[^>]*>");
-	private static final Pattern WHITESPACE_REGEXP = Pattern.compile("[\\s\\u00A0]");
-
 	@Inject
 	private Client client;
 
 	@Inject
 	private ChatMessageManager chatMessageManager;
 
+	@Inject
+	private ClientToolbar clientToolbar;
+
+	@Inject
+	private EmojiConfig config;
+
+	@Inject
+	private ClientThread clientThread;
+
+	static final String CONFIG_GROUP_KEY = "emoji";
+	private static final Pattern TAG_REGEXP = Pattern.compile("<[^>]*>");
+	private static final Pattern WHITESPACE_REGEXP = Pattern.compile("[\\s\\u00A0]");
+	private static final String EMIT = "Show";
+	private static final String MENU_TARGET = "Emoji Key";
+	private static final int EMOJI_EMIT_DURATION = 20;
+	private static final int EMOJI_PER_LINE = 8;
+	/**
+	 * queue of temporary messages added to the client
+	 */
+	private final Deque<EmojiEmitMessage> emojiEmitMessages = new ArrayDeque<>();
```

Comment:
I don't think we need to have multiples of these, if the player chooses to emit emojis in quick succession imo it's fine to just remove the current emitted message and send a new one.